### PR TITLE
Adds Plant Location Swiper functionality on Mobile

### PR DESCRIPTION
**Pull request by mohitb35**
- This PR adds the swiper layout to show plant locations and sample plant locations on mobile devices. 
- It also fixes a bug where the UI broke after exiting map mode on a project details screen, after selecting a plant location.
- The swiper layout is hidden on selecting a site, or exiting map mode. It is shown with appropriate details when a plant location or sample plant location is selected while in map mode.

Todo
- [x] General swiper layout
- [x] Plant location details swiper
- [x] Sample plant location swiper
- [x] Initialize popup - height issue
- [ ] ~Scroll issue on iOS safari when popup is open~ --> carried forward

### CODE REVIEW

**Comment by mariahosfeld** [comment]
- [ ] With Safari/iOS the font color of the link to show the whole plot again is blue instead of black.
Bildschirmfoto 2024-10-14 um 12 33 28
- [ ] When using the "Show whole Plot" link, the height of the swiper is not correct. Refresh solves this, swiping to the next slide as well
Bildschirmfoto 2024-10-14 um 12 34 51

**Comment by sunilsabatp** [comment]
Should we consider moving the satellite layer icon elsewhere because it is completely hidden in some cases, or should we disable this feature when a plant location is selected?

image

**Comment by sunilsabatp** [comment]
The expanded image mode for the plant location image slider is missing.
image

**Comment by mariahosfeld** [comment]
> The expanded image mode for the plant location image slider is missing. image

We did this by design, as it will not expand much more on mobile.

**Review by mariahosfeld — COMMENTED** [review]
The height issue with the slider seems to be resolved.
But now I am also facing the issue, that at least for the first sample tree I have to tap 2x to see the details.
The issue was also there in older deployments, but due to the now resolved height issue I did not notice when testing them.

**Comment by mohitb35** [comment]
> The height issue with the slider seems to be resolved. But now I am also facing the issue, that at least for the first sample tree I have to tap 2x to see the details. The issue was also there in older deployments, but due to the now resolved height issue I did not notice when testing them.

Need to debug that, seems to be related to event handling on mobile devices.

**Comment by mariahosfeld** [comment]
> > Should we consider moving the satellite layer icon elsewhere because it is completely hidden in some cases, or should we disable this feature when a plant location is selected?
> > image
> 
> I think we can hide the button in this scenario (on mobile devices).
> 
> @mariahosfeld thoughts?

I think so as well. On mobile we are either viewing the satellite layer or the plant locations.

**Review by sunilsabatp — CHANGES_REQUESTED** [review]

File: src/features/projectsV2/MobileInfoSwiper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import TreeMapperBrand from '../ProjectDetails/components/microComponents/TreeMapperBrand';
+import styles from './MobileInfoSwiper.module.scss';
+
+type MobileInfoSwiperProps = {
+  slides: ReactNode[];
+  uniqueKey: string;
+};
+
+function MobileInfoSwiper({ slides, uniqueKey }: MobileInfoSwiperProps) {
+  return (
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/features/projectsV2/MobileInfoSwiper/index.tsx:16]: I intentionally left it generic, so it could be used for any other purpose that we need.

**Comment by mohitb35** [comment]
> The height issue with the slider seems to be resolved. But now I am also facing the issue, that at least for the first sample tree I have to tap 2x to see the details. The issue was also there in older deployments, but due to the now resolved height issue I did not notice when testing them.

The 2x tap issue is now resolved.

**Review by sunilsabatp — APPROVED** [review]

**Pull request merged**